src/components/Input/Input.tsx

```diff
@@ -28,20 +30,29 @@ export const Input = ({ sticky }: { sticky: boolean }) => {
       }))
     );
 
-  const { handleSubmit } = useSubmitQuery();
-
   const inputRef = useRef<HTMLTextAreaElement>(null);
 
+  const handleSubmit = (promptApp: PromptApp) => {
+    trackEvent('HL Chat Message Sent', { 
+      messageLength: input.length,
+      hasAttachments: attachments.length > 0,
+      attachmentCount: attachments.length,
+      attachmentTypes: attachments.length > 0 ? attachments.map(a => a.type) : undefined
+    });
+    useSubmitQuery().handleSubmit(promptApp);
+    setInput("");
+  };
+
   const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
-    if (!inputIsDisabled && e.key === "Enter" && !e.shiftKey) {
+    if (!inputIsDisabled && e.key === "Enter" && !e.shiftKey && promptApp) {
```

Comment:
Is this intended? Will this only submit if there is a promptApp defined?

Reply:
Ah no -- good catch!

Reply:
Fixed